msamp/deepspeed/runtime/zero/fp8_stage_1_and_2.py

```diff
@@ -777,7 +797,12 @@ def all_gather_fp8_metas(self):
                 continue
             partition_size = len(params_partitions)
             scale_invs_partitions = [[p.meta.scale_inv for p in ps] for ps in params_partitions]
-            ref_scale = scale_invs_partitions[0][0]
+            ref_scale = None
+            for partition in scale_invs_partitions:
+                if len(partition) > 0:
+                    ref_scale = partition[0]
+                    break
+
```

Comment:
Please change the code to skip when ref_scale is None.
```python
if ref_scale is not None:
  scale_invs_partitions[pi].append(ref_scale.new_empty((pad, )))
```

Reply:
Sure, removed `ref_value`.